backend/src/router/account/non-fungible-tokens.ts

```diff
@@ -0,0 +1,170 @@
+  copies: number | null; // number of copies of this set of metadata in existence when token was minted.
+  issued_at: number | null; // When token was issued or minted, Unix epoch in milliseconds
+  expires_at: number | null; // When token expires, Unix epoch in milliseconds
+  starts_at: number | null; // When token starts being valid, Unix epoch in milliseconds
+  updated_at: number | null; // When token was last updated, Unix epoch in milliseconds
+  extra: string | null; // anything extra the NFT wants to store on-chain. Can be stringified JSON.
+  reference: string | null; // URL to an off-chain JSON file with more info.
+  reference_hash: string | null; // Base64-encoded sha256 hash of JSON from reference field. Required if `reference` is included.
+};
+
+// copied from Wallet https://github.com/near/near-wallet/blob/master/packages/frontend/src/services/NonFungibleTokens.js#L95
+const buildMediaUrl = (
+  media: TokenMetadata["media"],
+  base_uri: NFTContractMetadata["base_uri"]
+) => {
+  // return the provided media string if it is empty or already in a URI format
+  if (!media || media.includes("://") || media.startsWith("data:image")) {
+    return media;
+  }
+
+  if (base_uri) {
+    return `${base_uri}/${media}`;
+  }
+
+  return `https://cloudflare-ipfs.com/ipfs/${media}`;
+};
+
+export const router = trpc
+  .router<Context>()
+  .query("nonFungibleTokenContracts", {
+    input: z.strictObject({ accountId: validators.accountId }),
+    resolve: async ({ input: { accountId } }) => {
+      const selection = await indexerDatabase
+        .selectFrom("assets__non_fungible_token_events")
+        .select("emitted_by_contract_account_id as contractId")
+        .distinctOn("emitted_by_contract_account_id")
+        .where("token_new_owner_account_id", "=", accountId)
+        .execute();
+      return selection.map((row) => row.contractId);
+    },
+  })
+  .query("nonFungibleTokens", {
+    input: z.strictObject({
+      contractId: z.string(),
+      accountId: validators.accountId,
+      limit: validators.limit,
+      cursor: z.number().optional(),
+    }),
+    resolve: async ({
+      input: { contractId, accountId, limit, cursor = 0 },
+    }) => {
+      const nonFungibleTokenCount: string = await nearApi.callViewMethod(
+        contractId,
+        "nft_supply_for_owner",
+        {
+          account_id: accountId,
+        }
+      );
+      const nonFungibleTokenContractMetadata: NFTContractMetadata =
+        await nearApi.callViewMethod(contractId, "nft_metadata", {});
+
+      const nonFungibleTokenMetadata: Token[] = await nearApi.callViewMethod(
+        contractId,
+        "nft_tokens_for_owner",
+        {
+          account_id: accountId,
+          from_index: cursor.toString(),
+          limit,
+        }
+      );
+
+      return nonFungibleTokenMetadata.map((token, index) => ({
```

Comment:
We can now return `items` + `cursor` as an object (no need to extract `cursor` from last item), see transactions or beta account activity.